### PR TITLE
Libnetwork vendoring

### PR DESCRIPTION
**Pull request by fcrisciani**
Fixes in the Service Discovery areas

Signed-off-by: Flavio Crisciani <flavio.crisciani@docker.com>

**- What I did**
Libnetwork vendoring
Commits: https://github.com/docker/libnetwork/compare/f4a15a0890383619ad797b3bd2481cc6f46a978d...6426d1e66f33c0b0c8bb135b7ee547447f54d043

### CODE REVIEW

File: vendor/github.com/docker/libnetwork/service_linux.go [hunk 1]
```diff
@@ -520,7 +520,7 @@ func writePortsToFile(ports []*PortConfig) (string, error) {
 	}
 	defer f.Close()
 
-	buf, err := proto.Marshal(&EndpointRecord{
+	buf, _ := proto.Marshal(&EndpointRecord{
```
Comment (cpuguy83) [inline vendor/github.com/docker/libnetwork/service_linux.go]: Is ignoring this error intentional?

**Review by cpuguy83 — APPROVED** [review]
One question... but LGTM.

File: vendor/github.com/docker/libnetwork/service_linux.go (hunk 1 above)
Comment (fcrisciani) [inline vendor/github.com/docker/libnetwork/service_linux.go]: The err assignment was already not checked in original code

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Pull request merged**